modules/ding_mobilesearch/ding_mobilesearch.module

```diff
@@ -0,0 +1,504 @@
+  $tree = taxonomy_get_tree($vocabulary->vid, 0, NULL, TRUE);
+  $realm = array();
+  foreach ($tree as $term) {
+    if (!empty($term->field_agency_id[LANGUAGE_NONE][0]['value'])) {
+      $agency_id = $term->field_agency_id[LANGUAGE_NONE][0]['value'];
+      $realm[$agency_id] = $term->name . ' (' . $agency_id . ')';
+    }
+  }
+  if ($realm) {
+    $menu_name = $form['old_name']['#value'];
+    $form['agency_id'] = array(
+      '#type' => 'radios',
+      '#title' => t('Realm'),
+      '#description' => t('Please specify the realm for this menu.'),
+      '#options' => $realm,
+      '#default_value' => variable_get('mobilesearch_menu_agency_id__' . $menu_name, 0),
+      '#required' => TRUE,
+    );
+    $form['#submit'][] = 'ding_mobilesearch_menu_edit_submit';
+  }
+}
+
+/**
+ * Implements hook_form_BASE_ID_alter().
+ */
+function ding_mobilesearch_form_node_form_alter(&$form, &$form_state, $form_id) {
+  $plugin_info = module_invoke_all('mobilesearch_plugin_info');
+  foreach ($plugin_info as $row) {
+    if (!empty($row['node_form_callback']) && function_exists($row['node_form_callback'])) {
+      $row['node_form_callback']($form, $form_state);
+    }
+  }
+}
+
+/**
+ * Submit handler for a menu editing form.
+ */
+function ding_mobilesearch_menu_edit_submit($form, &$form_state) {
+  $menu_name = $form_state['values']['menu_name'];
+  variable_set('mobilesearch_menu_agency_id__' . $menu_name, $form_state['values']['agency_id']);
+}
+
+/**
+ * Returns list of menus with plugin associations.
+ */
+function _ding_mobilesearch_menus() {
+  $result = db_select('menu_custom', 'm')
+    ->fields('m', array('menu_name', 'title'))
+    ->orderBy('m.title')
+    ->execute();
+  $menus = array();
+  $plugin_info = module_invoke_all('mobilesearch_plugin_info');
+  foreach ($result as $row) {
+    if (preg_match('/^menu-og/', $row->menu_name)) {
+      continue;
+    }
+    $key = 'mobilesearch_menu__' . $row->menu_name;
+    if (variable_get($key, FALSE)) {
+      $plugin = variable_get($key . '__plugin', NULL);
+
+      $menus[$row->menu_name] = $plugin_info[$plugin];
+    }
+  }
+
+  return $menus;
+}
+
+/**
+ * Returns list of content types with plugin associations.
+ */
+function _ding_mobilesearch_types() {
+  $names = node_type_get_names();
+  $types = array();
+  $plugin_info = module_invoke_all('mobilesearch_plugin_info');
+  foreach ($names as $machine_name => $name) {
+    $key = 'mobilesearch_type__' . $machine_name;
+    if (variable_get($key, FALSE)) {
+      $trigger = variable_get($key . '__trigger', NULL);
+      $plugin = variable_get($key . '__plugin', NULL);
+
+      $types[$machine_name] = $plugin_info[$plugin];
+      $types[$machine_name]['trigger'] = $trigger;
+      $types[$machine_name]['plugin'] = $plugin;
+    }
+  }
+
+  return $types;
+}
+
+/**
+ * Implements hook_mobilesearch_node_export_mapping().
+ */
+function ding_mobilesearch_mobilesearch_node_export_mapping($node) {
+  $account = user_load($node->uid);
```

Comment:
This will load the profile with all fields and the users profile2 profile to get the users user name? This could be fetch much faster in other ways (SQL).